src/lib/WalletInfo.ts

```diff
@@ -201,8 +210,14 @@ export class WalletInfo {
                 internal: this.btcAddresses.internal.map((btcAddressInfo) => btcAddressInfo.toBtcAddressType()),
                 external: this.btcAddresses.external.map((btcAddressInfo) => btcAddressInfo.toBtcAddressType()),
             },
+            uid: await this.getUid(),
```

Comment:
The way the `uid` is generated and returned is different from how it's done for other properties like the account id.
Previously, `toAccountType` was just converting the data format of `WalletInfo` instances without adding or generating data.
All other data is previously generated and added to the wallet instance and also stored in the wallet store (also taking up a bit of storage space obviously).
So question is which approach we prefer.

Maybe @Yukiioz also has an opinion on this.

Reply:
The difference is that generating the UID is async, and all other data generation is sync. I would like to generate the UID also in the constructor already, but constructors cannot be async.

Reply:
For the account id, we pass the asynchronously pre-derived id into the constructor.